File: src/main/java/cn/huangsy/KidsWithCandies.java
package cn.huangsy;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

/**
 * @author huangsy
 * @date 2020/6/1 8:38
 */
public class KidsWithCandies {

    public List<Boolean> kidsWithCandies(int[] candies, int extraCandies) {
        List<Boolean> result = new ArrayList<>();
        int max = 0;
        for (int candy : candies) {
            max = Math.max(candy, max);
        }

        for (int candy : candies) {
            if (candy + extraCandies >= extraCandies) {
                result.add(true);
            } else {
                result.add(false);
            }
        }
        return result;
    }

    public static void main(String[] args) {
        KidsWithCandies kidsWithCandies = new KidsWithCandies();
        kidsWithCandies.kidsWithCandies(new int[]{7,2,5,6,9,9,4,5}, 3);
    }
}

File: src/main/java/cn/leetcode/1431/KidsWithCandies.kt
package cn.leetcode.`1431`

import java.util.*

/**
 * @author huangsy
 * @date 2020/6/1 8:36
 */
class KidsWithCandies {
    fun kidsWithCandies(candies: IntArray, extraCandies: Int): List<Boolean>? {
        val result: MutableList<Boolean> = ArrayList()
        var max = 0
        for (candy in candies) {
            max = Math.max(candy, max)
        }
        for (candy in candies) {
            if (candy + extraCandies >= extraCandies) {
                result.add(true)
            } else {
                result.add(false)
            }
        }
        return result
    }

}